packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -1867,6 +1867,48 @@ void main() {
       imageProvider2.complete();
       expect(tester.takeException(), isNull);
     });
+
+    testWidgets('Switch is using inactive track color when set', (WidgetTester tester) async {
+        const Color inactiveTrackColor = Color(0xFF00FF00);
+
+        await tester.pumpWidget(
+          const Directionality(
+            textDirection: TextDirection.ltr,
+            child: Center(
+              child: CupertinoSwitch(
+                value: false,
+                inactiveTrackColor: inactiveTrackColor,
+                dragStartBehavior: DragStartBehavior.down,
+                onChanged: null,
+              ),
+            ),
+          ),
+        );
+
+        expect(find.byType(CupertinoSwitch), findsOneWidget);
+        expect(tester.widget<CupertinoSwitch>(find.byType(CupertinoSwitch)).inactiveTrackColor, inactiveTrackColor);
+    });
+
+    testWidgets('Switch is using active track color when set', (WidgetTester tester) async {
+        const Color activeTrackColor = Color(0xFF00FF00);
+
+        await tester.pumpWidget(
+          const Directionality(
+            textDirection: TextDirection.ltr,
+            child: Center(
+              child: CupertinoSwitch(
+                value: false,
+                activeTrackColor: activeTrackColor,
```

Comment:
Even when the value is `false` it uses activeTrackColor? Maybe you should also test when value is `true`.